portal/src/graphql/adminapi/AddPhoneScreen.tsx

```diff
@@ -84,16 +107,61 @@ const AddPhoneForm: React.FC<AddPhoneFormProps> = function AddPhoneForm(
   }, [screenState, countryCodeConfig.default]);
 
   const onAddClicked = useCallback(() => {
-    // TODO: mutation to be integrated
-  }, []);
+    setDisableBlockNavigation(true);
+    const combinedPhone = `+${countryCode}${phone}`;
+    createIdentity({ key: "phone", value: combinedPhone })
+      .then((identity) => {
+        if (identity != null) {
+          navigate("../#connected-identities");
+        } else {
+          throw new Error();
```

Comment:
It seems we have two kinds of error handling:
- call the mutation function, and use `catch` on the result promise
- check for the `error` returned from `useMutation` hook

Can you help checking whether the first case will actually trigger? i.e. GraphQL errors would throw a JS error from the mutation function promise. If not, we can simplify and ignore it, and use `error` from mutation hook exclusively.

Reply:
> first case will actually trigger?

It will, so we can either use `.catch(() => {})` and use the error given be `useMutation` hook, or just use the error given by `catch`.

Reply:
So both of them are the same error instance? I prefer we use the error returned from the hook.

Reply:
> So both of them are the same error instance?

same instance, at least the data structure is exactly the same